PSReadLine/BasicEditing.cs

```diff
@@ -138,7 +141,8 @@ private static void ForwardDeleteImpl(int endPosition, Action<ConsoleKeyInfo?, o
                         !InViEditMode()));
 
                 buffer.Remove(current, length);
-                _singleton.Render();
+                // Moves cursor to current (backwards probably) and then deletes length
+                _singleton.SafeRender($"\x1b[{length}P", current);
```

Comment:
`ESC [ <n> P` only shifts characters from the current line, not the characters of the next line. So, this may not work for editing multi-line text.

Reply:
Yeah I should note in the PR description, this has not been tested with multi-line input at all.